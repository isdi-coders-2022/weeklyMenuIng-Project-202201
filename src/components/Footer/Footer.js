import styled from "styled-components";

const FooterContainer = styled.footer`
  display: flex;
  height: 100px;
  align-items: center;
  background-color: #b3dee2;
  justify-content: center;
  & p {
    padding: 0;
    margin: 0;
    font-size: 1.2rem;
    color: #151f28;
  }
`;

const Footer = () => {
  return (
    <FooterContainer>
      <p>My Weekly Menu</p>
    </FooterContainer>
  );
};
export default Footer;
