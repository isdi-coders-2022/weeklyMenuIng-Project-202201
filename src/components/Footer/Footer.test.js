import { render, screen } from "@testing-library/react";
import Footer from "./Footer";
import TestRenderer from "react-test-renderer";

describe("Given a Footer Component", () => {
  describe("When it's rendered", () => {
    test("Then it should render a footer element with text 'My Weekly Menu'", () => {
      const textFooter = "My Weekly Menu";
      render(<Footer />);

      const footerTextElement = screen.getByText(textFooter);

      expect(footerTextElement).toBeInTheDocument();
    });

    test("Then it should match the snapshot", () => {
      const footer = TestRenderer.create(<Footer />).toJSON();

      expect(footer).toMatchSnapshot();
    });
  });
});
