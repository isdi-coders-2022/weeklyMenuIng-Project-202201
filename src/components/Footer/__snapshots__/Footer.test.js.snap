// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Footer Component When it's rendered Then it should match the snapshot 1`] = `
<footer
  className="sc-bdvvtL cIVHyr"
>
  <p>
    My Weekly Menu
  </p>
</footer>
`;
